### PR TITLE
BNH-28

### PR DESCRIPTION
**Pull request by SleepyUser**
Refactored/restructured the test project to use inheritance to universalise setup for tests.
Additionally, workflow created to run all tests in test project on Pull Request, specifically when the following triggers occur: 
- "a user is assigned to the pull request", 
- "pull request is created", 
- "commit(s) pushed to the pull request", 
- "closed pull request is reopened"

### CODE REVIEW

**Review by emilypatsko — CHANGES_REQUESTED** [review]

**Review by BenRamchandani — CHANGES_REQUESTED** [review]
Some comments, hopefully all quick changes.

At some point we should make these warnings errors (i.e. `dotnet build -warnaserror`), but that can be another ticket.

File: .github/workflows/test_on_push_bricks-and-hearts.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Docs for the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
+# More GitHub Actions for Azure: https://github.com/Azure/actions
+
+name: Build and run tests on Pull Requests and Push commands
+
+on:
+  pull_request:
+    types: [ assigned, opened, synchronize, reopened]
```
[the remark this answers is not in the document]
Comment (emilypatsko) [reply .github/workflows/test_on_push_bricks-and-hearts.yml:8]: I said the same earlier, but figured it wasn't doing a whole lot of harm being there if there weren't other changes (a PR gets assigned to one of us automatically anyway 😄 )

File: BricksAndHearts.UnitTests/BricksAndHearts.UnitTests.csproj [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.11.0" />
+        <PackageReference Include="xunit" Version="2.4.1" />
+        <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
+            <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+            <PrivateAssets>all</PrivateAssets>
+        </PackageReference>
+        <PackageReference Include="coverlet.collector" Version="3.1.0">
+            <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+            <PrivateAssets>all</PrivateAssets>
+        </PackageReference>
```
[the remark this answers is not in the document]
Comment (emilypatsko) [reply BricksAndHearts.UnitTests/BricksAndHearts.UnitTests.csproj:22]: I don't recognise it... I recognise all the others though. It seems to be a code coverage tool - did you guys set this up?

[the remark this answers is not in the document]
Comment (SleepyUser) [reply BricksAndHearts.UnitTests/BricksAndHearts.UnitTests.csproj:22]: We didn't touch this file at all, not sure what's happening here. Should we remove it? @BenRamchandani @emilypatsko

[the remark this answers is not in the document]
Comment (BenRamchandani) [reply BricksAndHearts.UnitTests/BricksAndHearts.UnitTests.csproj:22]: Let's delete it if we're not using it, can always add it again

[the remark this answers is not in the document]
Comment (SleepyUser) [reply BricksAndHearts.UnitTests/BricksAndHearts.UnitTests.csproj]: Deleted :)

**Comment by SleepyUser** [comment]
Rebased and commit changes :)

**Review by BenRamchandani — APPROVED** [review]

**Pull request merged**